### DISCUSSION TITLE
How do I generate the insights page from my GitHub Actions?

### DISCUSSION THREAD

**Discussion by vittodevit** [lowlighter/metrics#568] [Q&A] [answered]
I currently have set up an action to generate my profile metrics. Recently I discovered the new Insights mode, with the full page thing. Can I generate the full page from my GitHub action to embed it on my website?

**Comment by lowlighter** [comment]
Hi!


Currently this is not supported as it's generated with VueJS from JSON output but I guess we could add a `html` output.

Not sure how it could be embedded though, I don't think it's possible directly from GitHub as it returns a mime type of `plain/text` to avoid people abusing raw.githubusercontent so it would require to proxy/cors it with new headers.

It would work when using GitHub pages though

**Reply by vittodevit** [reply]
Yeah of course, I understand. I think self hosting would be ok? If I self host in theory there will be no CORS problems. Correct me if I'm wrong.

**Comment by lowlighter** [comment, marked as answer]
This has been shipped with #575 and will available in next release 🎉

**Reply by vittodevit** [reply]
Fantastic, thank you!
